### ISSUE TITLE
autodoc tries to import classes as modules?

### ISSUE THREAD

**Issue by amueller** [sphinx-doc/sphinx#3768]
After updating sphinx and numpydoc, I get 

```
WARNING: /home/andy/checkout/scikit-learn/doc/modules/generated/sklearn.preprocessing.RobustScaler.rst:92: (WARNING/2) autodoc: failed to import method 'RobustScaler.__init__' from module 'sklearn.preprocessing.data.RobustScaler'; the following exception was raised:
Traceback (most recent call last):
  File "/home/andy/anaconda3/lib/python3.5/site-packages/sphinx/ext/autodoc.py", line 658, in import_object
    __import__(self.modname)
ImportError: No module named 'sklearn.preprocessing.data.RobustScaler'; 'sklearn.preprocessing.data' is not a package
```
for every class in the package.

It looks like the modname is wrong as it is the class and not the module. I'm not sure what should be imported at this point, though. The class? Importing a method doesn't seem to make a lot of sense...

**Comment by tk0miya** [comment]
Could you share your project to reproduce the error?
It would be nice if you provide small reproducible examples :-)

Thanks,

**Comment by amueller** [comment]
The project is scikit-learn. I'll try to extract a minimal example, but
I'll be traveling.

**Comment by tk0miya** [comment]
Hmm.. scikit-learn's doc is too large and complex to investigate...
I failed to do that today.

**Comment by goosemania** [comment]
@amueller , I suggest to test your case with this recently merged fix https://github.com/sphinx-doc/sphinx/pull/3780 . It is very small in terms of the changes to the code but it has a huge impact on how import works in certain cases. Hopefully it covers your case as well.

**Comment by amueller** [comment]
Thanks for looking into it. I'll have a look next week when I'm back from vacation ;)

**Comment by amueller** [comment]
The issue is still present on master.

**Comment by amueller** [comment]
I stripped down the sklearn build:
https://github.com/amueller/sphinx-debug

you need to add the doc folder to the pythonpath so the blablub.py is found as a module.

I get
```
WARNING: /home/andy/checkout/sphinx-debug/doc/modules/generated/blablub.BaseEstimator.rst:28: (WARNING/2) autodoc: failed to import method 'BaseEstimator.__init__' from module 'blablub.BaseEstimator'; the following exception was raised:
Traceback (most recent call last):
  File "/home/andy/checkout/sphinx/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
```

**Comment by tk0miya** [comment]
Yes, autodoc imports the target module internally on generating document.

It seems your example surely raises ImportError, but the error reason is different with first case. Is this intended?
```
WARNING: /docs/sphinx-debug/doc/modules/classes.rst:1: (WARNING/2) autodoc: failed to import module 'blablub'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 658, in import_object
    __import__(self.modname)
ImportError: No module named 'blablub'
```

The message on first case is here:
```
ImportError: No module named 'sklearn.preprocessing.data.RobustScaler'; 'sklearn.preprocessing.data' is not a package
```

**Comment by amueller** [comment]
@tk0miya as I said, you need to add the doc path to the python path for the module to be found. Not sure how to autodoc a module that's not installed.

**Comment by amueller** [comment]
anything I can do to make reproducing this easier? solving this will make our release process much easier, currently it prevents us from using stock packages for building our docs.

**Comment by tk0miya** [comment]
> Not sure how to autodoc a module that's not installed.

Please add your library's path to `sys.path` in `conf.py` like following:
```
import os
import sys

sys.path.insert(0, os.path.abspath("."))
```

**Comment by amueller** [comment]
done, not that that's any different from ``export PYTHONPATH=`pwd` ``

**Comment by jschueller** [comment]
maybe it was fixed by numpy/numpydoc#93 ?

**Comment by amueller** [comment]
@jschueller indeed it seems to be fixed on numpydoc master. Thanks!
